### PR TITLE
Supporting changes for a tanstack table implementation

### PR DESCRIPTION
**Pull request by reteps**
- URL utility
- Installation of packages
- Cleanup of ESLint comments

### CODE REVIEW

File: apps/prairielearn/src/lib/client/downloads.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * @param data 2D array of data.
+ * @returns A CSV-formatted string.
+ */
+export function arrayToCsv(data: unknown[][]): string {
+  return data
+    .map((row) =>
+      row
+        .map((field) => {
+          const stringField = String(field ?? '');
+          return /[",\n]/.test(stringField) ? `"${stringField.replace(/"/g, '""')}"` : stringField;
```
Comment (reteps) [inline apps/prairielearn/src/lib/client/downloads.ts]: This was written by Cursor, though it seems reasonable.

File: apps/prairielearn/src/lib/client/downloads.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        })
+        .join(','),
+    )
+    .join('\r\n');
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/client/downloads.ts]: Manual CSV construction makes me a little nervous, but I could just be overestimating how difficult it is to correctly and safely produce a CSV string.

Should we reach for [`csv-stringify/sync`](https://github.com/adaltas/node-csv/blob/master/packages/csv-stringify) instead to let someone else deal with the complexity? At the very least, I'd encourage you to check out their implementation to see if we're missing any edge cases. If you want to continue rolling your own, we should have tests for this.

Comment (reteps) [reply apps/prairielearn/src/lib/client/downloads.ts]: Done in [c22ef32](https://github.com/PrairieLearn/PrairieLearn/pull/12278/commits/c22ef322227ecf117893a62f616be40bb4aa7c6f).
